src/pages/DetailsPage.jsx

```diff
@@ -1,15 +1,28 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import Heading from "../components/Heading";
 import Breadcrumbs from "../components/Breadcrumb";
 import Input from "../components/Input";
 import Select from "../components/Select";
 import CurrentDate from "../components/Date";
 
 function DetailsPage() {
-  const options = [
-    { currency: "elo", label: "elo" },
-    { currency: "siema", label: "siema" },
-  ];
+  const [list, setList] = useState("");
+
+  useEffect(() => {
+    fetch(
+      ` https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/latest/currencies/eur.json`,
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setList(Object.keys(data.eur));
+      });
+  }, []);
+  const array = [...list];
+  const options = [];
+  array.forEach((value, index) => {
+    options.push({ currency: list[index], label: list[index] });
+  });
```

Comment:
This implementation needs to be improved to incorporate the following:
- separating the logic in a custom hook
- handling loading state
- handling error state

I have a small refactor plan for you about this part.

1. Move the `list` state and the `useEffect` to a custom hook `useCurrenciesList` and use the hook in the page (you can define the hook above the component) - the functionality  should remain the same
2. Introduce 2 new states in the hook `loading` (boolean) and `error` (boolean)
3. Before fetching in the `useEffect` set the loading state to `true` and when fetching is done, set it to `false`.
4. Using `try catch` block set error to `true` in the `catch` phrase, by default it should be `false`.

As a side note, use `.map` not `.forEach` for this operation.
In the diff, I showed you how to hook should be used

```suggestion
  // single source of truth for fetching the list, the component is easy to read
  const { list, loading, error } = useCurrenciesList();
  // no foreach, using the value directly instead of list[index], not defining redundant constants
  const options = list.map((item) => ({ currency: item, label: item }));
```

also in the page conditionally show loader if `loading` is `true` and show an error message if `error` is `true`

Reply:
done